packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2273,7 +2328,19 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     );
   }
 
-  public isEip1559Active(): boolean {
+  public isEip1559Active(blockNumberOrPending?: BN | "pending"): boolean {
+    const block =
+      blockNumberOrPending !== "pending" ? blockNumberOrPending : undefined;
+    if (
+      block !== undefined &&
+      this._forkBlockNumber !== undefined &&
+      block.lt(new BN(this._forkBlockNumber))
+    ) {
+      return this._vm._common.hardforkGteHardfork(
+        this._selectHardfork(block),
+        "london"
+      );
+    }
     return this._vm._common.gteHardfork("london");
```

Comment:
I think this can also be simplified if _selectHardfork also had the "is historic" condition inside

Reply:
I confirmed that _selectHardfork has sufficient "is historic" checks, and then removed the redundant "is historic" checks in https://github.com/nomiclabs/hardhat/pull/1928/commits/f4b28f5c0ddd6304171f3267b64982e8b00c8f97